<launch>
  <!-- Scene topic arguments-->
  <arg name="background_pcl2_topic"          default="/costar_sp_segmenter/table_corner" />
  <arg name="scene_pcl2_topic"               default="/costar_sp_segmenter/points_out" />
  <arg name="detected_object_topic"          default="/costar_sp_segmenter/detected_object_list"/>
  <arg name="object_hypotheses_topic"        default="/costar_sp_segmenter/object_hypothesis" />

  <arg name="object_folder_location"         default="$(find sequential_scene_parsing)/mesh" />
  <arg name="background_normal_as_gravity"   default="true"/>
  <arg name="background_mode"                default="0"/>
  <arg name="TF_z_inv_gravity_dir"           default="/table_frame"/>
  <arg name="tf_publisher_initial"           default="scene"/>
  <arg name="debug"                          default="false"/>
  <arg name="load_table"                     default="true"/>
  <arg name="table_location"                 default="$(find object_on_table_segmenter)/data/table.pcd"/>
  <arg name="render_scene"                   default="true" />
  <arg name="objransac_model_directory"      default="$(find sequential_scene_parsing)/mesh"/>
  <arg name="objransac_model_names"          default="wood_cube" />

  <arg name="data_forces_magnitude"          default="2.0"/>
  <arg name="data_forces_max_distance"       default="0.05"/>
  <arg name="data_forces_model"              default="2"/>

  <arg name="best_hypothesis_only"           default="false"/>
  <arg name="small_obj_g_comp"               default="2"/>
  <arg name="sim_freq_multiplier"            default="3."/>

  <!-- physics solver settings: check http://bulletphysics.org/mediawiki-1.5.8/index.php/BtContactSolverInfo -->
  <arg name="p_solver_iter"                  default="20"/>
  <arg name="p_randomize_order"              default="true"/>
  <arg name="p_split_impulse"                default="false"/>
  <arg name="p_penetration_threshold"        default="-0.02"/>

  <node pkg="sequential_scene_parsing" type="sequential_scene_ros" name="sequential_scene_parsing"
  output="screen" 
  >  
    <!-- sequential_scene_ros common Node arg pass -->
    <param name="small_obj_g_comp"        type="int"     value="$(arg small_obj_g_comp)"/>
    <param name="sim_freq_multiplier"     type="double"  value="$(arg sim_freq_multiplier)"/>

    <param name="background_pcl2_topic"   type="str"     value="$(arg background_pcl2_topic)" />
    <param name="scene_pcl2_topic"        type="str"     value="$(arg scene_pcl2_topic)" />

    <param name="object_folder_location"  type="str"     value="$(arg object_folder_location)"/>
    <param name="detected_object_topic"   type="str"     value="$(arg detected_object_topic)"/>
    <param name="TF_z_inv_gravity_dir"    type="str"     value="$(arg TF_z_inv_gravity_dir)"/>
    <param name="tf_publisher_initial"    type="str"     value="$(arg tf_publisher_initial)"/>
    <param name="object_hypotheses_topic" type="str"     value="$(arg object_hypotheses_topic)"/>

    <param name="debug_mode"              type="bool"    value="$(arg debug)"/>
    <param name="load_table"              type="bool"    value="$(arg load_table)"/>
    <param name="background_mode"         type="int"     value="$(arg background_mode)"/>
    <param name="table_location"          type="str"     value="$(arg table_location)"/>
    <param name="bg_normal_as_gravity"    type="bool"    value="$(arg background_normal_as_gravity)"/>

    <param name="render_scene"            type="bool"    value="$(arg render_scene)"/>
    <param name="best_hypothesis_only"    type="bool"    value="$(arg best_hypothesis_only)"/>

    <param name="p_solver_iter"            type="int"    value="$(arg p_solver_iter)"/>
    <param name="p_randomize_order"        type="bool"     value="$(arg p_randomize_order)"/>
    <param name="p_split_impulse"          type="bool"     value="$(arg p_split_impulse)"/>
    <param name="p_penetration_threshold"  type="double"    value="$(arg p_penetration_threshold)"/>

    <param name="data_forces_magnitude"        type="double"  value="$(arg data_forces_magnitude)"/>
    <param name="data_forces_max_distance"     type="double"  value="$(arg data_forces_max_distance)"/>
    <param name="data_forces_model"            type="int"     value="$(arg data_forces_model)"/>

    <param name="objransac_model_directory"    type="str"     value="$(arg objransac_model_directory)"/>
    <param name="objransac_model_names"        type="str"     value="$(arg objransac_model_names)"/>

    <!-- objectDatabase -->
    <rosparam command="load" file="$(find sequential_scene_parsing)/mesh/block_property_database.yaml" />
    
  </node>

</launch>
